Add endpoint to refresh auth token

Rushi1109 opened the pull request:
**Changes**
- Add a new endpoint to hit, when the auth token is expired.
- Add logic in controller to refresh the auth token.

Completes the backend part for #915

marcelluscaio commented on Server/controllers/authController.js:
```diff
@@ -173,6 +173,68 @@ const loginUser = async (req, res, next) => {
+      const error = new Error(errorMessages.NO_REFRESH_TOKEN);
+      error.status = 401;
+      error.service = SERVICE_NAME;
+      error.method = "handleExpiredJwtToken";
```
Here we do not have a refreshToken. should the method be handleExpiredJwtToken?

ajhollid replied:
`error.method` is meant to refer to the method that is throwing the error so we have more information in the logs to quickly track down where errors are being thrown.

In this case the method name should be `refreshAuthToken`

Rushi1109 replied:
> `error.method` is meant to refer to the method that is throwing the error so we have more information in the logs to quickly track down where errors are being thrown.
> 
> In this case the method name should be `refreshAuthToken`

Yeah. I am aware of that, I copied the code from the middleware, so forgot to update the method name there. I'll update it.

ajhollid reviewed (changes requested):
Looks pretty good to me for the most part :+1: 

The `error.method` value should be the name of the method that is throwing the error, so that ca be updated.

Please include updated tests for `authController` and document the new endpoint in the `openapi.json` file so we have complete documentation and tests.

Rushi1109 commented:
> Please include updated tests for `authController` and document the new endpoint in the `openapi.json` file so we have complete documentation and tests.

Hello @ajhollid,
I tried adding test cases but some lines are uncovered. I am having difficulties, so I would need some help here.

ajhollid commented:
> > Please include updated tests for `authController` and document the new endpoint in the `openapi.json` file so we have complete documentation and tests.
> 
> Hello @ajhollid, I tried adding test cases but some lines are uncovered. I am having difficulties, so I would need some help here.

Hi @Rushi1109,

I'd be happy to help with some test cases, but I would need you to grant me access to your forked repo so I can push to this branch.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Rushi1109 commented:
> I'd be happy to help with some test cases, but I would need you to grant me access to your forked repo so I can push to this branch.

Hello @ajhollid,

![image](https://github.com/user-attachments/assets/cefa42d4-941c-4a97-8948-0fb80d59515c)
I guess It was enabled by default.

ajhollid commented:
> > I'd be happy to help with some test cases, but I would need you to grant me access to your forked repo so I can push to this branch.
> 
> Hello @ajhollid,
> 
> ![image](https://github.com/user-attachments/assets/cefa42d4-941c-4a97-8948-0fb80d59515c)
> I guess It was enabled by default.

OK ill give it a try when I have a chance. It's the weekend coming up so it might be a couple days. I will get to it as soon as I can 👍

Rushi1109 commented:
> OK ill give it a try when I have a chance. It's the weekend coming up so it might be a couple days. I will get to it as soon as I can 👍

Sure.

ajhollid commented:
Hey @Rushi1109 ,

I refactored the function slightly, I'm not sure why all code following verifying refresh token was carried out in the callback, but I don't think there's any reason to. Please review the changes I made and make sure everything is still functioning as you expect.

I also added the missing test cases so you've got 100% coverage again now.

ajhollid commented:
@Rushi1109 do you still want to finish this PR?  I will close it otherwise.

Thanks!

Rushi1109 commented:
Hello @ajhollid, 
Sorry for the inactivity.

> I also added the missing test cases so you've got 100% coverage again now.
Thanks for adding those test cases.

> I refactored the function slightly, I'm not sure why all code following verifying refresh token was carried out in the callback, but I don't think there's any reason to. Please review the changes I made and make sure everything is still functioning as you expect.
Yeah I guess it should not be inside the callback. Thanks for correction.

I have reviewed the changes and it looks good to me.

Rushi1109 commented:
I think the PR is ready to be merged now.

@ajhollid.

ajhollid commented:
> Hello @ajhollid, 
> Sorry for the inactivity.
> 
> > I also added the missing test cases so you've got 100% coverage again now.
> 
> Thanks for adding those test cases.
> 
> > I refactored the function slightly, I'm not sure why all code following verifying refresh token was carried out in the callback, but I don't think there's any reason to. Please review the changes I made and make sure everything is still functioning as you expect.
> 
> Yeah I guess it should not be inside the callback. Thanks for correction.
> 
> I have reviewed the changes and it looks good to me.

No problem, just wanted to make sure you were still with us 😂

ajhollid commented:
> I think the PR is ready to be merged now.
> 
> @ajhollid.

Awesome thanks for your work! I will review as soon as I can 👍

Rushi1109 commented:
> No problem, just wanted to make sure you were still with us 😂

Yeah, It was a weekend. So, I wasn't working on it.

ajhollid reviewed (approved):
Looks good to me, thanks for making revisions :+1:
